.github/workflows/website-deploy-preview.yml

```diff
@@ -0,0 +1,71 @@
+name: "Website Deploy Preview"
+
+on:
+  workflow_dispatch:
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - "website/**"
+
+env:
+  AWS_REGION: us-east-2
+  IAM_ROLE_ARN: arn:aws:iam::068007702576:role/cplive-plat-ue2-dev-atmos-docs-github-action
+  IAM_ROLE_SESSION_NAME: cloudposse-atmos-ci-deploy-pr-${{ github.event.pull_request.number }}
+  S3_BUCKET_NAME: cplive-plat-ue2-dev-atmos-docs-origin
+  PR_NUMBER: ${{ github.event.pull_request.number }}
+
+# These permissions are needed to interact with the GitHub's OIDC Token endpoint
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  website-deploy-preview:
+    # Do not deploy the website to the preview environment if the PR has the label 'website-no-deploy'
+    if: ${{ !contains(github.event.*.labels.*.name, 'website-no-deploy') }}
+
+    runs-on: ubuntu-latest
+    steps:
+      # https://github.com/marketplace/actions/configure-aws-credentials-action-for-github-actions
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.IAM_ROLE_ARN }}
+          role-session-name: ${{ env.IAM_ROLE_SESSION_NAME }}
+
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: "website/.nvmrc"
+
+      - name: Install Dependencies and Build Website
+        run: |
+          node --version
+          cd website
+          npm install --only=production
+          npm run build:site
+
+      - name: Copy Website to S3 Bucket PR Folder
+        run: |
+          cd website/build
+          aws sts get-caller-identity
```

Comment:
Is this left over from iterative development? (i.e. checking that it's assuming the correct role before issuing the `aws s3 ...` commands).

Reply:
just for debugging - since all the infra for it is provisioned in another repo, and there are many resources involved, and if something happens, at least we'll see that the GitHub OIDC provider can't assume the correct role.
there are no secrets in there